pkg/sql/logictest/testdata/logic_test/vectorize

```diff
@@ -1291,3 +1291,28 @@ CREATE TYPE greeting AS ENUM ('hello');
 CREATE TABLE greeting_table (x greeting);
 EXPLAIN (VEC) SELECT * FROM greeting_table;
 SET vectorize = experimental_always
+
+# Regression test for #66706. Ensure that cfetcher can correctly determine when
+# consecutive KVs in a unique secondary index belong to the same row.
+statement ok
+SET vectorize = experimental_always
```

Comment:
I added this because I believe it is necessary in 20.2 to ensure that the queries below use the vectorized engine.